### PR TITLE
fix box lookup

### LINKED ISSUE #60: Mempool: input boxes and proofBytes can sometimes return null
When we use the following query as example, we can sometimes get an empty box in an input or get null proofBytes.

`query Transactions {
  mempool {
    transactions {
      transactionId
      size
      timestamp
      inputs {
        box {
          boxId
          value
          address
        }
        proofBytes
      }
    }
  }
}`
image
image

### PR DESCRIPTION
**Pull request by SepehrGanji**
fix #60

### CODE REVIEW

File: src/context/unconfirmed-transactions-repository.ts [hunk 1]
```diff
@@ -66,6 +67,32 @@ export class UnconfirmedTransactionRepository extends BaseRepository<Unconfirmed
 
       return filterQuery;
     });
+
+    const u_box_ids: string[] = [];
+    records.forEach((record) => {
+      record.inputs.forEach((input) => {
```
Comment (capt-nemo429) [inline src/context/unconfirmed-transactions-repository.ts]: Query will fail if `inputs` isn't selected. So better to have a check if isn't empty or undefined before to proceed.

File: src/context/unconfirmed-transactions-repository.ts [hunk 2]
```diff
@@ -66,6 +67,32 @@ export class UnconfirmedTransactionRepository extends BaseRepository<Unconfirmed
...
+        if (input.box === null) u_box_ids.push(input.boxId);
+      });
+    });
+    const u_boxes = await this.dataSource
```
Comment (capt-nemo429) [inline src/context/unconfirmed-transactions-repository.ts]: We can avoid unnecessary database query and post data processing by first checking if `u_box_ids` isn't empty.

**Review by capt-nemo429 — CHANGES_REQUESTED** [review]
Great work! Left some comments on the code. 

One remark: Let's keep the code style consistent by using camelCase. :)

File: src/context/unconfirmed-transactions-repository.ts (hunk 1 above)
Comment (SepehrGanji) [reply src/context/unconfirmed-transactions-repository.ts]: Done, nice catch )

File: src/context/unconfirmed-transactions-repository.ts (hunk 2 above)
Comment (SepehrGanji) [reply src/context/unconfirmed-transactions-repository.ts]: Refactored.

**Review by capt-nemo429 — APPROVED** [review]

**Pull request merged**